service/history/workflow/mutable_state_impl.go

```diff
@@ -4601,6 +4601,17 @@ func (ms *MutableStateImpl) closeTransactionWithPolicyCheck(
 	}
 }
 
+func (ms *MutableStateImpl) BufferSizeAcceptable() bool {
+	if ms.hBuilder.NumBufferedEvents() >= ms.config.MaximumBufferedEventsBatch() {
```

Comment:
I think we can change this to `>`, so that when someone set the config to be 2, it really mean 2 event can be buffered, not 1...
should be safe for existing deployment as well as we are basically allowing one more event.

Reply:
Fixed for both.